scripts/Synapse/convertParquetMPI.ipynb

```diff
@@ -1,152 +1,171 @@
+    "        return AccessToken(\n",
+    "            token=mssparkutils.credentials.getToken(audience=\"vault\"),\n",
+    "            expires_on=int(time.time())+60*10 # some random time in future... Synapse doesn't document how to get the actual time\n",
+    "        )\n",
+    "\n",
+    "credential = ManagedIdentityCredential()\n",
+    "credential._credential = spoof_token() # monkey-patch the contents of the private `_credential`\n",
+    "\n",
+    "# Set up key vault client\n",
+    "vault_name = \"$KEY_VAULT\"\n",
+    "vault_url = f\"https://{vault_name}.vault.azure.net/\"\n",
+    "\n",
+    "client = SecretClient(vault_url=vault_url, credential=credential)\n",
+    "\n",
+    "# Get client ID and secret for GitHub app registration\n",
+    "client_id = client.get_secret(\"synapse-client-id\").value\n",
+    "client_secret = client.get_secret(\"synapse-client-secret\").value\n",
+    "\n",
+    "# Get access token for record linkage container app\n",
+    "tenant_id = \"$TENANT_ID\"\n",
+    "url = f\"https://login.microsoftonline.com/{tenant_id}/oauth2/v2.0/token\"\n",
+    "\n",
+    "data = {\n",
+    "    'grant_type': 'client_credentials',\n",
+    "    'client_id': client_id,\n",
+    "    'client_secret': client_secret,\n",
+    "    'scope': 'api://phdi-dev-record-linkage/.default'\n",
+    "}\n",
+    "\n",
+    "response = requests.post(url, data=data)\n",
+    "access_token = response.json()['access_token']\n",
+    "\n",
+    "# Make request to record linkage container app\n",
+    "record_linkage_url = client.get_secret(\"record-linkage-url\") + \"/link-record\"\n",
+    "ingestion_url = client.get_secret(\"ingestion-url\")\n",
+    "std_names_url = ingestion_url + \"/standardize_names\"\n",
+    "std_phones_url = ingestion_url + \"/standardize_phones\"\n",
+    "std_address_url = ingestion_url + \"/geocode_bundle\"\n",
+    "headers = {\n",
+    "    'Authorization': f'Bearer {access_token}'\n",
+    "}\n"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Read the MPI parquet data into a spark dataframe. Iterate over each row of patient data in the dataframe and convert to a FHIR bundle and associated iris_id. Create a POST request to the record linkage container with FHIR bundle and iris_id."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Convert and post data from mpi_incoming_file_path\n",
+    "def convert(mpi_incoming_file_path):\n",
+    "    df = spark.read.parquet(mpi_incoming_file_path)\n",
+    "    for row in df.collect():\n",
+    "        iris_id, fhir_bundle = convert_to_patient_fhir_resources(row.asDict())\n",
+    "\n",
+    "        # Add metadata to LAC Extract denoting it came from IRIS\n",
+    "        # We know this URI is ~probably~ not right but Brady might know what to use...\n",
+    "        fhir_bundle[\"meta\"] = {\"source\": \"uri:iris\"}\n",
```

Comment:
Based upon what I have read up about this, I think this should suffice for now.  Later we can use an actual [Provenance](https://www.hl7.org/fhir/provenance.html) resource with additional information if we wanted.  We may want to add something to the source uri to indicate LAC if this was going to be reused by any other STLT.  

Some information from HL7 about the Meta.Source element:
Definition:	
A uri that identifies the source system of the resource. This provides a minimal amount of Provenance information that can be used to track or differentiate the source of information in the resource. The source may identify another FHIR server, document, message, database, etc.
Comments:
The exact use of the source (and the possible implied Provenance.entity.role and agent.role) is left to implementer discretion. Only one nominated source is allowed; for additional provenance details, a full Provenance resource should be used. The source may correspond to Provenance.entity.what[x] or Provenance.agent.who[x], though it may be a more general or abstract reference.